application/src/main/data/upgrade/3.8.1/schema_update.sql

```diff
@@ -24,3 +24,139 @@ UPDATE user_credentials c SET failed_login_attempts = (SELECT (additional_info::
+                ON CONFLICT DO NOTHING;
+                -- create bundle for android and iOS app
+                generatedBundleId := uuid_generate_v4();
+                INSERT INTO mobile_app_bundle(id, created_time, tenant_id, title, android_app_id, ios_app_id, oauth2_enabled)
+                VALUES (generatedBundleId, (extract(epoch from now()) * 1000), mobileAppRecord.tenant_id,
+                        'Autogenerated for ' || mobileAppRecord.pkg_name, mobileAppRecord.id, iosAppId, mobileAppRecord.oauth2_enabled)
+                ON CONFLICT DO NOTHING;
+                UPDATE mobile_app_bundle_oauth2_client SET mobile_app_bundle_id = generatedBundleId WHERE mobile_app_bundle_id = mobileAppRecord.id;
+            END LOOP;
+        END IF;
+        ALTER TABLE mobile_app DROP COLUMN IF EXISTS oauth2_enabled;
+        IF NOT EXISTS(SELECT 1 FROM pg_constraint WHERE conname = 'pkg_name_platform_unique') THEN
+            ALTER TABLE mobile_app ADD CONSTRAINT pkg_name_platform_unique UNIQUE (pkg_name, platform_type);
+        END IF;
+    END;
+$$;
+
+ALTER TABLE IF EXISTS mobile_app_settings RENAME TO qr_code_settings;
+ALTER TABLE qr_code_settings ADD COLUMN IF NOT EXISTS mobile_app_bundle_id uuid;
+
+-- migrate mobile apps from qr code settings to mobile_app, create mobile app bundle for the pair of apps
+DO
+$$
+    DECLARE
+        iosPkgName varchar;
+        androidAppId uuid;
+        iosAppId uuid;
+        generatedBundleId uuid;
+        qrCodeRecord RECORD;
+    BEGIN
+        -- in case of running the upgrade script a second time
+        IF EXISTS(SELECT 1 FROM information_schema.columns WHERE table_name = 'qr_code_settings' AND column_name = 'android_config') THEN
+            FOR qrCodeRecord IN SELECT * FROM qr_code_settings
+            LOOP
+                generatedBundleId := NULL;
+                -- migrate android config
+                SELECT id into androidAppId FROM mobile_app WHERE pkg_name = qrCodeRecord.android_config::jsonb ->> 'appPackage' AND platform_type = 'ANDROID';
+                IF androidAppId IS NULL THEN
+                    androidAppId := uuid_generate_v4();
+                    INSERT INTO mobile_app(id, created_time, tenant_id, pkg_name, platform_type, status, store_info)
+                    VALUES (androidAppId, (extract(epoch from now()) * 1000), qrCodeRecord.tenant_id,
+                            qrCodeRecord.android_config::jsonb ->> 'appPackage', 'ANDROID', 'DRAFT', qrCodeRecord.android_config::jsonb - 'appPackage');
+                    generatedBundleId := uuid_generate_v4();
+                    INSERT INTO mobile_app_bundle(id, created_time, tenant_id, title, android_app_id)
+                    VALUES (generatedBundleId, (extract(epoch from now()) * 1000), qrCodeRecord.tenant_id, 'Autogenerated for qr code', androidAppId);
```

Comment:
Maybe we should use pkg name as title? Just pkg name or "pkg_name (autogenerated)". Same for the script above (line 91)